api/python/provisioner/commands/cluster_id.py

```diff
@@ -21,22 +21,77 @@
 import uuid
 from typing import Type
 
+from . import (
+     CommandParserFillerMixin
+)
 from .. import inputs
 from ..api import grains_get
-from ..config import LOCAL_MINION
+from ..config import (
+     LOCAL_MINION,
+     ALL_MINIONS
+)
+from ..vendor import attr
 
 from provisioner.commands import (
-     PillarSet,
-     GetClusterId
+     PillarGet,
+     PillarSet
 )
-from . import CommandParserFillerMixin
-
-from ..vendor import attr
 from provisioner.salt import local_minion_id
 
 logger = logging.getLogger(__name__)
 
 
+@attr.s(auto_attribs=True)
+class GetClusterId(CommandParserFillerMixin):
+    input_type: Type[inputs.NoParams] = inputs.NoParams
+
+    @staticmethod
+    def run(targets=ALL_MINIONS):
+        """get_cluster_id command execution method.
+
+        Gets cluster ID from pillar data.
+
+        Execution:
+        `provisioner get_cluster_id`
+        Takes no mandatory argument as input.
+
+        """
+        try:
+            cluster_data = PillarGet().run(
+                'cluster',
+                targets=targets
+            )
+            cluster_id = []
+
+            for node in cluster_data:
+                if "cluster_id" in cluster_data[node]["cluster"]:
```

Comment:
Does it handle None value for key **_cluster_id_**?
Key could be set with a None value, in which case we might end up appending None or a blank value.